test/utils/test_bit_register_mapping.py

```diff
@@ -0,0 +1,71 @@
+import math
+from collections.abc import Iterable
+
+import pytest
+
+from opensquirrel import CircuitBuilder
+from opensquirrel.default_instructions import measure
+from opensquirrel.ir import Measure
+from opensquirrel.utils.bit_register_mapping import BitStringMapping
+
+
+@pytest.fixture(name="bit_string_mapping")
+def bit_string_mapping_fixture() -> BitStringMapping:
+    return BitStringMapping()
+
+
+@pytest.mark.parametrize(
+    ("measurements", "expected_mapping", "expected_bit_allocation"),
+    [
+        ([measure(0, 0)], {0: [0]}, {0: 0}),
+        ([measure(0, 0), measure(1, 0)], {0: [0], 1: [0]}, {0: 1}),
+        ([measure(0, 1), measure(0, 0)], {0: [1, 0]}, {0: 0, 1: 0}),
+    ],
+)
+def test_add_measure(
+    bit_string_mapping: BitStringMapping,
+    measurements: Iterable[Measure],
+    expected_mapping: dict[int, list[int]],
```

Comment:
Can we use a `BitStringMapping` for expected_mapping instead of a `dict[int, list[int]]`?

Reply:
Arguably we shouldn't test private methods, private properties, just public methods. Anyway, my point is that any complex type exposed from the class should be simple to read. And aliases are good for that.